api/v1beta3/vcdcluster_types.go

```diff
@@ -80,39 +79,112 @@ type LoadBalancerConfig struct {
+	// corresponding OVDC is created. There will be a separate layer that is pre-created and managed by the customer
+	// which can multiplex across the load-balancers.
+	ExternalLoadBalancer ZoneTopologyType = "ExternalLoadBalancer"
+)
+
+// DCGroupConfig defines configuration for DCGroup zone topology.
+type DCGroupConfig struct {
+	// TODO: decide on appropriate configuration for this type, or remove this struct
+}
+
+// UserSpecifiedEdgeGatewayConfig defines configuration for UserSpecifiedEdgeGateway zone topology.
+type UserSpecifiedEdgeGatewayConfig struct {
+	// EdgeGatewayZone defines the name of the user-provided zone containing an edge gateway.
+	EdgeGatewayZone string `json:"edgeGatewayZone"`
+}
+
+// ExternalLoadBalancerConfig defines configuration for ExternalLoadBalancer zone topology.
+type ExternalLoadBalancerConfig struct {
+	// EdgeGatewayZones defines a list of zones containing an edge gateway.
+	EdgeGatewayZones EdgeGatewayZones `json:"edgeGatewayZones"`
+}
+
+// EdgeGatewayZones defines a list of zones containing an edge gateway. For use with ExternalLoadBalancer topology.
+type EdgeGatewayZones []EdgeGatewayZone
+
+// EdgeGatewayZone provides the details of a zone containing an edge gateway. Used with ExternalLoadBalancer topology.
+// topology.
+type EdgeGatewayZone struct {
+	// Name is the user-defined name of the zone representing the OVDC containing an edge gateway.
+	Name string `json:"name"`
+	// LoadBalancerIP is the IP address of the load balancer configured on the edge gateway for this zone.
+	LoadBalancerIP string `json:"loadBalancerIP"`
+	// LoadBalancerPort is the port of the load balancer configured on the edge gateway for this zone.
+	LoadBalancerPort int `json:"loadBalancerPort"`
+}
+
 // Zone is an Availability Zone in VCD
 type Zone struct {
-	Name            string `json:"name"`
-	OVDCName        string `json:"ovdcName"`
+	// Name defines the name of this zone.
+	Name string `json:"name"`
+	// OVDCName defines the actual name of the OVDC which corresponds to this zone.
+	OVDCName string `json:"ovdcName"`
+	// OVDCNetworkName defines the OVDC network for this zone.
 	OVDCNetworkName string `json:"ovdcNetworkName"`
-
+	// ControlPlaneZone defines whether a control plane node can be deployed to this zone.
 	//+kubebuilder:default=false
 	ControlPlaneZone bool `json:"controlPlaneZone"`
+}
 
+// MultiZoneSpec provides details of the zone configuration in the cluster as well as the NetworkTopology used. Only one
+// of DCGroupConfig, UserSpecifiedEdgeGatewayConfig, or ExternalLoadBalancerConfig should be provided corresponding to the
+// chosen ZoneTopology.
+type MultiZoneSpec struct {
+	// ZoneTopology defines the type of network topology used across zones in a Multi-AZ deployment.
+	// Valid options are DCGroup, UserSpecifiedEdgeGateway, and ExternalLoadBalancer,
 	// +optional
-	LoadBalancerIP string `json:"loadBalancerIP,omitempty"`
+	ZoneTopology ZoneTopologyType `json:"ZoneTopology,omitempty"`
```

Comment:
the json representation should start with lowercase (`zoneTopologyType`) in accordance with k8s conventions.